feat: add hide root to agnoster short

jedwillick opened the pull request:
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

Adds the `hide_root_location` toggle to Agnoster Short preventing it from displaying the root directory if it exceeds the `max_depth`. 

With `hide_root_location` true and `max_depth` of 2:
![image](https://user-images.githubusercontent.com/85419773/156113600-8775e115-7f63-4d07-8cb1-7d4fac7fe671.png)


Kind Regards,
Jed

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

jedwillick committed:
feat: add hide root to agnoster short

JanDeDobbeleer commented on src/segments/path.go:
```diff
@@ -274,8 +274,26 @@ func (pt *Path) getFullPath() string {
+	if pathDepth < maxDepth {
+		return pt.getFullPath()
+	}
+	pathSeparator := pt.env.PathSeparator()
```
this and the above code (`getAgnosterShortPath`) is almost identical, can we find a way to reuse it?

jedwillick commented:
I fixed the repeated logic but now im kinda thinking that maybe there should just be an option in agnoster_short to remove always showing the root folder. What are your thoughts?

JanDeDobbeleer commented:
@jedwillick I would try it 😃

jedwillick commented:
Updated PR. Added an `hide_root` option to `agnoster_short` instead of chagning `folder`.

JanDeDobbeleer commented on docs/docs/segment-path.md:
```diff
@@ -36,6 +36,8 @@ Display the current path.
 - mixed_threshold: `number` - the maximum length of a path segment that will be displayed when using `Mixed` -
   defaults to `4`
 - max_depth: `number` - maximum path depth to display before shortening when using `Agnoster Short` - defaults to `1`
+- hide_root: `boolen` -  hides the root directory if it doesn't fit in the last `max_depth` folders, when using
```
- typo: `boolean`
- needs to be `agnoster_short`

JanDeDobbeleer commented on src/segments/path.go:
```diff
@@ -56,6 +56,8 @@ const (
 	// MaxDepth Maximum path depth to display whithout shortening
 	MaxDepth properties.Property = "max_depth"
+	// Hides the root folder if it doesn't fit in max_depth. Used in Agnoster Short
+	HideRoot properties.Property = "hide_root"
```
I would rename to `hide_root_location` as it can be ambiguous.

JanDeDobbeleer commented on src/segments/path.go:
```diff
@@ -249,18 +251,27 @@ func (pt *Path) getAgnosterShortPath() string {
-	if pathDepth <= maxDepth {
+	hideRoot := pt.props.GetBool(HideRoot, false)
+	if !hideRoot {
+		pathDepth--
```
this needs a comment as it's a bit magic once the context of this PR is gone.

jedwillick commented:
Made requested changes.

- [x] Docs: Fixed typo (`boolen` -> `boolean`)
- [x] Docs: Changed `Agnoster Short` -> `agnoster_short`
- [x] All: Changed `hide_root` -> `hide_root_location`
- [x] Path: Added comment to the magic part and flipped logic to hopefully make it more clear aswell
- [x] Path Test: Also added a few more tests

JanDeDobbeleer commented:
@jedwillick on a roll :-) Awesome tbh.
